libs/lexer/src/Lexer.php

```diff
@@ -169,19 +188,19 @@ private function run(ReadableInterface $source, int $offset): \Generator
                 continue;
             }
 
-            if ($token->getName() === $this->driver::UNKNOWN_TOKEN_NAME) {
+            if (!$this->possibleTokensSearching && $token->getName() === $this->driver::UNKNOWN_TOKEN_NAME) {
                 $unknown[] = $token;
                 continue;
             }
 
-            if (\count($unknown) && $token->getName() !== $this->driver::UNKNOWN_TOKEN_NAME) {
+            if ($this->throwOnError && \count($unknown) && $token->getName() !== $this->driver::UNKNOWN_TOKEN_NAME) {
                 throw UnrecognizedTokenException::fromToken($source, $this->reduceUnknownToken($unknown));
             }
 
             yield $token;
```

Comment:
There's an error now. The response is not buffered.

That is, this means that in the case:

```php
$lexer = new Lexer(['T_NUMBER' => '\d+', 'T_PLUS' => '\+', 'T_WS' => '\s+'], ['T_WS']);
$lexer->lex('2 + err 2');

// Expected:
- "2" T_NUMBER
- "+" T_PLUS
- "err" T_UNKNOWN
- "2" T_NUMBER

// Actual:
- "2" T_NUMBER
- "+" T_PLUS
- "e" T_UNKNOWN
- "r" T_UNKNOWN
- "r" T_UNKNOWN
- "+" T_PLUS
- "2" T_NUMBER
```

Reply:
Hmm... Although it's not a fact that this is a bug, it seems to me that this is more expected behavior, isn't it (TBD)?

Reply:
It's a bug or unexpected behavior. I fixed it. I added reduceUnknownToken for Lexer working without UnknownTokenException and renamed Lexer method to disableUnrecognizedTokenException.